src/AuthManager.ts

```diff
@@ -88,9 +122,8 @@ export class AuthManager {
                 if (decodedToken) {
                     const currentTime = Date.now() / 1000;
                     if (decodedToken.exp < currentTime) {
-                        // add refresh check here instead and
-                        localStorage.removeItem('access_token');
-                        return resolve(false);
+                        // add refresh check here
+                        this.refreshAccessToken();
```

Comment:
refresh token also has expiry time, check expiry of refresh token before trying to use it